src/pages/availability-panel.tsx

```diff
@@ -1,55 +1,72 @@
 import Head from '@/components/head';
 import Layout from '@/components/Layout';
-import useFetch from '@/hooks/useFetch';
 import task from '@/interfaces/task.type';
 import classNames from '@/styles/availabilityPanel.module.scss';
+import fetch from '@/helperFunctions/fetch';
 import DragDropcontext from '../components/availability-panel/drag-drop-context/index';
 
-const TASKS_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/tasks`;
-const IDLE_MEMBERS_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/members/idle`;
-
 const AvailabilityPanel: FC = () => {
   const [idleMembersList, setIdleMembersList] = useState<string[]>([]);
-  const [tasks, setTasks] = useState<task[]>([]);
   const [unAssignedTasks, setUnAssignedTasks] = useState<task[]>([]);
-
-  const {
-    response: taskResponse,
-    error: taskError,
-    isLoading: taskIsLoading,
-  } = useFetch(TASKS_URL);
-
-  const { response, error, isLoading } = useFetch(IDLE_MEMBERS_URL);
+  const [error, setError] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [reRenderComponent, setReRenderComponent] = useState<boolean>(false);
 
   useEffect(() => {
-    if ('tasks' in taskResponse) {
-      setTasks(taskResponse.tasks);
-      const active = tasks.filter(
-        (item: task) => item.status.toLowerCase() === 'unassigned',
-      );
-      setUnAssignedTasks(active);
-    }
-
-    if ('idleMemberUserNames' in response) {
-      const filterMembers = response.idleMemberUserNames.filter((username:string) => username);
-      const sortedIdleMembers = filterMembers.sort();
-      setIdleMembersList(sortedIdleMembers);
-    }
-  }, [isLoading, response, taskIsLoading, taskResponse]);
+    const fetchTasks = async () => {
+      try {
+        setIsLoading(true);
+        const url = `${process.env.NEXT_PUBLIC_BASE_URL}/tasks`;
+        const response = await fetch({ url });
+        if (response.status === 200) {
+          const { tasks } = response.data;
+          const unassigned = tasks.filter(
+            (item: task) => item.status.toLowerCase() === 'unassigned' && item.type === 'feature',
+          );
+          setUnAssignedTasks(unassigned);
+        }
+      } catch (Error) {
+        setError(true);
+        setIsLoading(false);
+      }
+    };
+    const fetchIdleUsers = async () => {
+      try {
+        setIsLoading(true);
+        const url = `${process.env.NEXT_PUBLIC_BASE_URL}/members/idle`;
+        const response = await fetch({ url });
+        if (response.status === 200) {
+          const { idleMemberUserNames } = response.data;
+          const filterMembers = idleMemberUserNames.filter((username:string) => username);
+          const sortedIdleMembers = filterMembers.sort();
+          setIdleMembersList(sortedIdleMembers);
+        }
+        setIsLoading(false);
+      } catch (Error) {
+        setError(true);
+        setIsLoading(false);
+      }
+    };
+    fetchTasks();
```

Comment:
Should I make this IIFE?

Reply:
No, it's good.
Just move the cleanup like setting loading false and clearing the errors in `finally` block